### PR TITLE
[DOM-380] - Implement domains endpoint

### PR DESCRIPTION
**Pull request by Manu0912**
## Summary
- Domains endpoint was implemented
- Tests were implemented
### Related links:
- [Task DOM-380](https://makingsense.atlassian.net/jira/software/projects/DOM/boards/84?selectedIssue=DOM-380)
- https://github.com/FromDoppler/doppler-push-contact/pull/149

### CODE REVIEW

File: Doppler.PushContact/Services/PushContactService.cs [hunk 1]
```diff
@@ -265,6 +267,79 @@ public async Task<IEnumerable<string>> GetAllDeviceTokensByDomainAsync(string do
...
+                    .Select(x =>
+                    {
+                        return new DomainInfo()
+                        {
+                            Name = x.GetValue(PushContactDocumentProps.DomainPropName, null)?.AsString,
+                            PushContactActiveQuantity = x.GetValue("PushContactActiveQuantity", null).ToInt32(),
+                            PushContactInactiveQuantity = x.GetValue("PushContactInactiveQuantity", null).ToInt32()
+                        };
+                    })
+                    .ToList();
```
Comment (andresmoschini) [inline Doppler.PushContact/Services/PushContactService.cs:331]: Very minor, but I think that it is less verbose and clear

```suggestion
                var domainList = domainsFiltered
                    .Select(x => new DomainInfo()
                    {
                        Name = x.GetValue(PushContactDocumentProps.DomainPropName, null)?.AsString,
                        PushContactActiveQuantity = x.GetValue("PushContactActiveQuantity", null).ToInt32(),
                        PushContactInactiveQuantity = x.GetValue("PushContactInactiveQuantity", null).ToInt32()
                    })
                    .ToList();
```

File: Doppler.PushContact/Controllers/PushContactController.cs [hunk 2]
```diff
@@ -159,9 +159,20 @@ public async Task<ActionResult<ApiPage<MessageDeliveryResult>>> GetMessages([Fro
...
-        public async Task<ActionResult<ApiPage<DomainInfo>>> GetDomains()
+        public async Task<ActionResult<ApiPage<DomainInfo>>> GetDomains(int page, int per_page)
         {
-            throw new NotImplementedException();
+            if (page < 0)
+            {
+                return BadRequest($"'{nameof(page)}' cannot be lesser than 0.");
+            }
+
+            if (per_page <= 0 || per_page > 100)
```
Comment (andresmoschini) [inline Doppler.PushContact/Controllers/PushContactController.cs:169]: Very minor, but maybe you can define a constant for 100.

File: Doppler.PushContact/Controllers/PushContactController.cs [hunk 3]
```diff
@@ -159,9 +159,20 @@ public async Task<ActionResult<ApiPage<MessageDeliveryResult>>> GetMessages([Fro
...
+            {
+                return BadRequest($"'{nameof(per_page)}' has to be greater than 0 and lesser than 100.");
```
Comment (andresmoschini) [inline Doppler.PushContact/Controllers/PushContactController.cs:171]: [The same here](https://github.com/FromDoppler/doppler-push-contact/pull/151/files#r924463488)

File: Doppler.PushContact/ApiModels/ApiPage.cs [hunk 4]
```diff
@@ -16,5 +16,11 @@ public ApiPage(List<T> items, int page, int per_page)
             Page = page;
             Per_page = per_page;
         }
+
+        public ApiPage(int page, int per_page)
+        {
+            Page = page;
+            Per_page = per_page;
+        }
```
Comment (andresmoschini) [inline Doppler.PushContact/ApiModels/ApiPage.cs:24]: I think that this constructor is not being used.

**Review by leoslopez — APPROVED** [review]
👍

File: Doppler.PushContact/Services/PushContactService.cs [hunk 5]
```diff
@@ -265,6 +267,79 @@ public async Task<IEnumerable<string>> GetAllDeviceTokensByDomainAsync(string do
...
 
+        public async Task<ApiPage<DomainInfo>> GetDomains(int page, int per_page)
+        {
+            try
+            {
+                var domainsFiltered = await PushContacts.Aggregate()
+                    .Group(new BsonDocument
+                    {
+                        { "_id", $"${PushContactDocumentProps.DomainPropName}" },
+                        { "PushContactActiveQuantity",
```
Comment (leoslopez) [inline Doppler.PushContact/Services/PushContactService.cs:278]: Very minor too, but I would organize/indent the code of both properties, `PushContactActiveQuantity` and `PushContactInactiveQuantity`, in equal way, in order to understand/remember the solution when look the code again :D

**Pull request merged**